src/Workspaces/Core/Portable/EmbeddedLanguages/LanguageServices/EmbeddedLanguageClassifierContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+{
+    internal struct EmbeddedLanguageClassifierContext
+    {
+        internal readonly ClassificationOptions Options;
+        private readonly ArrayBuilder<ClassifiedSpan> _result;
+
+        /// <summary>
+        /// The string or character token to classify.
+        /// </summary>
+        public SyntaxToken SyntaxToken { get; }
+
+        /// <summary>
+        /// SemanticModel that <see cref="SyntaxToken"/> is contained in.
+        /// </summary>
+        public SemanticModel SemanticModel { get; }
+
+        public CancellationToken CancellationToken { get; }
+
+        internal EmbeddedLanguageClassifierContext(
```

Comment:
Why internal members? Is this going to be promoted to public APIs?

Reply:
It will be once we have confidence through a third party :-)